modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt

```diff
@@ -304,3 +307,14 @@ interface SequenceKSemialign : Semialign<ForSequenceK>, SequenceKFunctor {
 interface SequenceKAlign : Align<ForSequenceK>, SequenceKSemialign {
   override fun <A> empty(): Kind<ForSequenceK, A> = emptySequence<A>().k()
 }
+
+@extension
+interface SequenceKUnalign : Unalign<ForSequenceK>, SequenceKSemialign {
+  override fun <A, B> unalign(ior: Kind<ForSequenceK, Ior<A, B>>): Tuple2<Kind<ForSequenceK, A>, Kind<ForSequenceK, B>> =
+    ior.fix().let { seq ->
+      val ls = seq.sequence.filterMap { it.toLeftOption() }.k()
+      val rs = seq.sequence.filterMap { it.toOption() }.k()
+
+      ls toT rs
```

Comment:
This can be done in one pass as well, but don't use fold or methods from the foldable class, as this should be a lazy operation :/ (Those methods really need fixing, but thats not too easy)

Reply:
Actually nvm